requirements/base.txt

```diff
@@ -1,14 +1,14 @@
 # the default package dependencies
-
+certifi>=14.05.14
 hydra-core>=1.0.0
+lightly_utils~=0.0.0
 numpy>=1.18.1
-pytorch_lightning>=1.0.4,<1.7  # TODO: Hotfix, remove upper bound
+python_dateutil>=2.5.3
+pytorch_lightning>=1.5.0
```

Comment:
And this might restrict some users?

Reply:
I verified that 1.5.0 was before the lowest `pytorch_lightning` version that lightly would run with. So this line just states that explicitly.

Reply:
Hmm, that's weird. Let's double check this before we merge. I'm not a big fan of adding strict dependencies except if it's really required. 
We have some projects here that use much older versions: https://github.com/lightly-ai/lightly/network/dependents?package_id=UGFja2FnZS0xNDQ3MzY0NTYy

E.g. this one here uses lightly 1.2.27 with pytorch lightning 1.2.7 (https://github.com/huaxiuyao/Wild-Time/blob/main/requirements.txt)

Reply:
I did more investigation.

I managed to create an environment where I can install pytorch lightning 1.4. There are three tests that are failing (master branch):
```
FAILED tests/cli/test_cli_magic.py::TestCLIMagic::test_magic_without_upload_with_trainer - TypeError: __init__() got an unexpected keyword...
FAILED tests/cli/test_cli_train.py::TestCLITrain::test_train_weights_summary - TypeError: __init__() got an unexpected keyword argument 's...
FAILED tests/core/test_Core.py::TestCore::test_train_and_embed - TypeError: __init__() got an unexpected keyword argument 'strategy'
```

The error is:
```
lightly/embedding/_base.py:88: in train_embedding
    trainer = pl.Trainer(**kwargs, callbacks=[self.checkpoint_callback])
...
>       return fn(self, **kwargs)
E       TypeError: __init__() got an unexpected keyword argument 'strategy'
```

Reply:
@IgorSusmelj I see no great options here.
* We cannot support all the versions of pytorch_lightning because they make breaking changes too fast. Currently, cli does not work with versions before 1.5. We don't specify this lower bound in requirements.txt, so someone might install the package and it will not work.
* By specifying the lower bound, projects that use lightly with an older pytorch lightning version (which means that they don't use the broken part) will need to do dependency resolution.

I do think that the better option here is to specify the lower bound, even though some people will need to do dependency resolution.
* We are still supporting older versions (1.5+), not just the newest one (1.7)
* It is always possible to override and call `pip install` with an older version after lightly is installed
* I checked the list, there are just three that use lightly with and older pl version. I tried to install one locally (the one you mentioned). The dependencies though could not be resolved (I tried several python versions, ran `pip install -r requirements.txt`). Also they say it is a WIP.
* We have to draw a line somewhere. It seems to me that 1.5 is a reasonable version.

Reply:
Discussed with Igor in person.

Conclusion:
* We don't include the lower bound
* Create a [follow-up issue](https://github.com/lightly-ai/lightly/issues/912)